src/content-helper/common/utils/date.ts

```diff
@@ -13,20 +13,14 @@ const DATE_NOT_AVAILABLE_MESSAGE = __( 'Date N/A', 'wp-parsely' );
  * @param {Date} date The date to be examined.
  * @return {boolean} Whether the date can be processed further.
  */
-export function canProcessDate( date: Date|string ): boolean {
-	if ( 'string' === typeof date ) {
-		// Verify that date format is "YYYY-MM-DD".
-		if ( ! /^\d{4}-\d{2}-\d{2}$/.test( date ) ) {
-			return false;
-		}
-
-		date = new Date( date );
+export function canProcessDate( date: Date ): boolean {
+	// Return false if the date is not a valid Date object.
+	if ( isNaN( +date ) ) {
+		return false;
 	}
 
-	const isValidDateObject = date instanceof Date && ! isNaN( +date );
-	const isNotEpochDate = 0 !== date.getTime();
-
-	return isValidDateObject && isNotEpochDate;
+	// Return false if the date is the Unix Epoch time.
+	return 0 !== date.getTime();
```

Comment:
The comment is not entirely true because `new Date( 86400000 )` will pass this check.

I think we should remove this condition also because not sure what value it is adding.

Reply:
> The comment is not entirely true because new Date( 86400000 ) will pass this check.

Unless I misunderstood your comment, `new Date( 86400000 )` doesn't resolve to the epoch time.

> I think we should remove this condition also because not sure what value it is adding.

Tested a bit, and actually removing this will also allow invalid dates like `new Date( '2023-03-32' )`. I think that the current implementation is a good enough compromise. JS is just weird with dates.